### ISSUE TITLE
reportDeprecated not changed if TypeCheckingMode set to standard

### ISSUE THREAD

**Issue by MalcolmMielle** [DetachHead/basedpyright#1315]
### Description

Changing reportDeprecated to true doesn't show any warning:

```python
from typing import Iterator  #this is deprecated

def test():
    pass
```

With pyproject:

```toml
[tool.basedpyright]
typeCheckingMode = "standard"
reportDeprecated = "hint"
```

or 

```toml
[tool.basedpyright]
typeCheckingMode = "standard"
reportDeprecated = true
```

**Comment by DetachHead** [comment]
Deprecated diagnostics on builtin types are enabled with the `deprecateTypingAliases` setting

**Comment by MalcolmMielle** [comment]
That's right, it works now thanks!
